Sources/Features/Breadcrumb/BacktraceBreadcrumbFileHelper.swift

```diff
@@ -52,18 +52,21 @@ enum BacktraceBreadcrumbFileHelperError: Error {
         }
 
         let textBytes = Data(text.utf8)
-        if textBytes.count > BacktraceBreadcrumbFileHelper.maximumBreadcrumbSize {
-            BacktraceLogger.warning("We should not have a breadcrumb this big, this is a bug! Discarding breadcrumb.")
+        if textBytes.count > maximumIndividualBreadcrumbSize {
+            BacktraceLogger.warning(
+                "Discarding breadcrumb that was larger than the maximum specified (\(maximumIndividualBreadcrumbSize).")
             return false
         }
 
         do {
-            // Keep removing until there's enough space to add the new breadcrumb
-            while (queueByteSize() + textBytes.count) > maxQueueFileSizeBytes {
-                try queue.pop(1, error: ())
-            }
+            try dispatchQueue.sync {
+                // Keep removing until there's enough space to add the new breadcrumb (leaving 512 bytes room)
+                while (queueByteSize() + textBytes.count) > (maxQueueFileSizeBytes - 512) {
+                    try queue.pop(1, error: ())
```

Comment:
Correct me if I'm wrong but in swift/objective-c passing a reference to error object will automatically set an error if an error occurred. With that being said - wouldn't it be better if we instead using try/catch block use an exception reference? 
In the documentation:
https://github.com/linkedin/cassette
I also see that the pop/add method returns boolean - which tells me if we could add a text to the queue. Wouldn't it be better to use boolean instead of try/catch block? Right now try/catch block is useless

Reply:
This is part of the ObjC/Swift magic.

```
Objective-C methods that return BOOL and return an NSError by reference are exposed to Swift as methods that use the try/catch error-handling mechanism, and return Void.
```

https://stackoverflow.com/questions/46626571/how-to-convert-objective-c-bool-to-swifts-bool-with-nserror-as-one-of-the-par